api/src/main/java/org/apache/iceberg/actions/RewriteDeletes.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Set;
+import org.apache.iceberg.DeleteFile;
+
+public interface RewriteDeletes extends SnapshotUpdate<RewriteDeletes, RewriteDeletes.Result> {
+
+  /**
+   * Set the implementation class name for rewrite strategy.
+   *
+   * @return this for method chaining
+   */
+  RewriteDeletes strategy(String strategyImpl);
```

Comment:
Is it possible for us to pass in `RewriteDeleteStrategy` here rather than a free formed string?

Reply:
`RewriteDeleteStrategy` is defined in the core package, while this is in the api package. This is like the pattern for the `Catalog`, we use the API like this:

```java
Catalog catalog = CatalogUtil.loadCatalog(HiveCatalog.class.getName(), "hive", ImmutableMap.of(), hadoopConf);
```

Reply:
For data file rewrites, we didn't allow custom extension because we don't think that there's a lot of value in it just yet and we want to keep things simple if there isn't a use for dynamically loaded classes. I think we should probably take the same approach here.

Reply:
I think we all agree that eq->pos is the one that can share with others. Let me update this.

Reply:
+1 here too. We debated this while designing the new data compaction API.

Reply:
I think another common strategy would be to just compact delete files. For example, take 100 position deletes and write them as one.

Reply:
+1. That is the next action, I have verified internally that it could bring some benefit for query.